qa/rpc-tests/excessive.py

```diff
@@ -12,17 +12,231 @@
+            n+=group
+            if n >= len(addrs): n=0
+	  logging.info("mine blocks")
+          self.nodes[0].generate(5)  # mine all the created transactions
+	  logging.info("sync all blocks and mempools")
+	  self.sync_all()
+
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+
+          logging.info("Test not relaying a large transaction")
+
+  	  mbefore = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+          assert_equal(mbefore,[(0, 0)]*4)  # we need the mempool to be empty to track that this one tx doesn't prop
+
+  	  (tx, vin, vout, txid) = split_transaction(self.nodes[0],wallet[0:3000],[addrs[0]],txfeePer=60)
+          logging.debug("Transaction Length is: ", len(binascii.unhexlify(tx)))
+          assert(binascii.unhexlify(tx) > 100000) # txn has to be big for the test to work
+        
+  	  mbefore = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+          assert_equal(mbefore[1:],[(0, 0), (0, 0), (0, 0)])  # verify that the transaction did not propagate
+          assert(mbefore[0][0] > 0) # verify that the transaction is in my node
+
+          logging.info("Test a large transaction in block < 1MB")
+          largeBlock = self.nodes[0].generate(1)
+          self.sync_blocks()
+          counts = [ x.getblockcount() for x in self.nodes ]
+          latest = counts[0]
+          assert_equal(counts, [latest,latest,latest,latest]) # Verify that all nodes accepted the block, even if some of them didn't have the transaction.  They should all accept a <= 1MB block with a tx <= 1MB
+
+  	  # mafter = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+
+        if self.extended:  # this test checks the behavior of > 1MB blocks with excessive transactions.  it takes a LONG time to generate and propagate 1MB+ txs.
+  	  # Create a LOT of UTXOs for the next test
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+          logging.info("Create lots of UTXOs...")
+          n=0
+
+          for w in wallet:
+            split_transaction(self.nodes[0], [w], addrs[n:100+n])
+            n+=100
+            if n >= len(addrs): n=0
+	
+          self.nodes[0].generate(1)
+	  self.sync_all()
+
+          logging.info("Building > 1MB block...")
+          self.nodes[0].set("net.excessiveTxn=1000000")  # Set the excessive transaction size larger for this node so we can generate an "excessive" block for the other nodes
+        
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=False)
+
+          # Generate 1 MB worth of transactions        
+          size = 0
+          count = 0
+          while size < 1000000:
+            count+=1
+            utxo = wallet.pop()
+            outp = {}
+	    outp[addrs[count%len(addrs)]] = utxo["amount"]
```

Comment:
@gandrewstone I know you are using emacs so this is could worth considering in regards of indentation/formatting: https://www.emacswiki.org/emacs/IndentingPython 
The same apply for C++ code: https://www.emacswiki.org/emacs/CPlusPlusMode#toc1